chemprop/train/train.py

```diff
@@ -68,13 +74,23 @@ def train(model: MoleculeModel,
         targets = targets.to(torch_device)
         target_weights = target_weights.to(torch_device)
         data_weights = data_weights.to(torch_device)
+        if args.loss_function == 'bounded_mse':
+            lt_target_batch = lt_target_batch.to(torch_device)
+            gt_target_batch = gt_target_batch.to(torch_device)
 
         # Calculate losses
-        if args.dataset_type == 'multiclass':
+        if args.loss_function == 'mcc' and args.dataset_type == 'classification':
+            loss = loss_func(preds, targets, data_weights, mask) *target_weights.squeeze(0)
+        elif args.loss_function == 'mcc': # multiclass dataset type
+            targets = targets.long()
+            loss = torch.cat([loss_func(preds[:, target_index, :], targets[:, target_index], data_weights[:, target_index], mask[:, target_index]) for target_index in range(preds.size(1))], dim=1) * target_weights.squeeze(0)
```

Comment:
This is an important catch. Multiclass is not currently included in the integration testing. And the manual testing I did was just on the function and not on this aggregation step. Thanks for the catch and the fix!

Reply:
I tried it with a test file of my own that had two tasks for the multiclass job. And it revealed another problem. To find the source of the dimensionality problem, I had to break up the multiclass loss into multiple lines. And since I've always kind of thought the multiclass loss line was compact but unreadable, I unfolded it for the regular multiclass case too.